ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -64,15 +74,55 @@ public static boolean isPseudoColumn(final ColumnName columnName) {
     return pseudoColumnNames().contains(columnName);
   }
 
+  public static Set<ColumnName> pseudoColumnNames(final int pseudoColumnVersion) {
+    if (!PseudoColumns.PSEUDO_COLUMN_NAMES_BY_VERSION.containsKey(pseudoColumnVersion)) {
+      throw new KsqlException("Provided pseudoColumnVersion has no corresponding columns defined");
+    }
+    return PseudoColumns.PSEUDO_COLUMN_NAMES_BY_VERSION.get(pseudoColumnVersion);
+  }
+
   public static Set<ColumnName> pseudoColumnNames() {
-    return PSEUDO_COLUMN_NAMES;
+    return pseudoColumnNames(CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
   }
 
   public static boolean isSystemColumn(final ColumnName columnName) {
     return systemColumnNames().contains(columnName);
   }
 
   public static Set<ColumnName> systemColumnNames() {
-    return SYSTEM_COLUMN_NAMES;
+    return systemColumnNames(CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
+  }
+
+  public static Set<ColumnName> systemColumnNames(final int pseudoColumnVersion) {
+    return SYSTEM_COLUMN_NAMES_BY_VERSION.get(pseudoColumnVersion);
+  }
+
+  private static Set<ColumnName> buildColumns(final int pseudoColumnVersion) {
+    return ImmutableSet.<ColumnName>builder()
+        .addAll(pseudoColumnNames(pseudoColumnVersion))
+        .addAll(WINDOW_BOUNDS_COLUMN_NAMES)
+        .build();
+  }
+
+  static class PseudoColumns {
```

Comment:
Make this class private and add an accessor method to expose looking up specific entries in the `PSEUDO_COLUMN_NAMES_BY_VERSION` map? Essentially, move the logic that's currently in `pseudoColumnNames()` into this class in order for it to be more contained. If this class owns the map, it'd be nice for it to own how the map is accessed too.